Author: Jonathan Riddell <jr@jriddell.org>
Date:   Thu Oct 9 10:35:49 2014 +0200
Description: Update default applications in kickoff
    instead of defaulting to konqueror find the first browser from a list
    change konsole to ktp-contactlist, we want to appeal to non geeks
    change kmail for kontact, we provide a lovely PIM suite so we should show it by default
Forwarded: https://git.reviewboard.kde.org/r/120538/

diff --git a/applets/kickoff/core/favoritesmodel.cpp b/applets/kickoff/core/favoritesmodel.cpp
index e26cf80..e1d3a71 100644
--- a/applets/kickoff/core/favoritesmodel.cpp
+++ b/applets/kickoff/core/favoritesmodel.cpp
@@ -109,7 +109,18 @@ public:
     static QList<QString> defaultFavorites()
     {
         QList<QString> applications;
-        applications << "konqbrowser" << "KMail2" << "systemsettings" << "dolphin" << "org.kde.konsole" << "org.kde.kwrite";
+
+        QList<QString> browsers;
+        browsers << "konqbrowser" << "rekonq" << "firefox" << "chromium-browser" << "google-chrome";
+        foreach (const QString& browser, browsers) {
+            KService::Ptr service = KService::serviceByStorageId(browser + ".desktop");
+            if (service) {
+                applications << browser;
+                break;
+            }
+        }
+
+        applications << "kontact" << "systemsettings" << "dolphin" << "ktp-contactlist" << "kwrite";
 
         QList<QString> desktopFiles;
 
